4-WebApp-your-API/4-3-AnyOrg/AppCreationScripts/sample.json

```diff
@@ -15,37 +15,42 @@
       "Id": "service",
       "Name": "WebApi-MultiTenant-v2",
       "Kind": "WebApi",
-      "Audience": "AzureADMultipleOrgs",
       "HomePage": "https://localhost:44351/",
       "ReplyUrls": "https://localhost:44351/api/Home",
+      "Audience": "AzureADandPersonalMicrosoftAccount",
       "PasswordCredentials": "Auto",
       "RequiredResourcesAccess": [
         {
           "Resource": "Microsoft Graph",
           "DelegatedPermissions": [ "User.Read.All" ]
         }
+      ],
+      "Scopes": [ "Read.User.Data", "Write.User.Data" ],
```

Comment:
>Read.User.Data [](http://example.com/codeflow?start=19&length=14)

Since the codegen generates good standard text very well for ToDoList.* scopes and app permissions, I'd suggest you change the AP permissions to map 4-1 ..
then you can copy the ToDoListController code as well from 4-1